pvlib/modelchain.py

```diff
@@ -589,8 +672,9 @@ def dc_model(self, model):
             model = model.lower()
             if model in _DC_MODEL_PARAMS.keys():
                 # validate module parameters
-                missing_params = (_DC_MODEL_PARAMS[model]
-                                  - set(self.system.module_parameters.keys()))
+                missing_params = (
+                    _DC_MODEL_PARAMS[model] -
+                    _array_keys(self.system.module_parameters, 0))
```

Comment:
The idea is that we have (in `ModelChain.__init__`) made sure that all arrays have the same keys in their `module_parameters` and `temperature_model_parameters` so if we want to infer a model, or check that we can apply a model given what we know about the system we just need to look at the keys for the first array.

I'm glad you brought this up because I wrote it thinking it would be much more useful and general. As it stands `_array_keys()` is only ever called with `array=0` so at a minimum that parameter should go away. The best thing might be to replace it with a function that returns the intersection of the set of keys for each dict in a tuple.

Reply:
> The best thing might be to replace it with a function that returns the intersection of the set of keys for each dict in a tuple.

I like that. The important thing isn't that the keys are the same, it's that all of the relevant keys are present for each model.